feat(examples): added unit test condition example

dhutchison opened the pull request:
I kind of got sidetracked with other things and haven't really got around to working on #249 anymore since my last comment on it. On the upside though, over the last few weeks adopting unit testing CloudFormation with this in my team has went down very well & its giving me a few ideas for little usability improvements that could be made. 

Instead of trying to complete all the examples in one PR though I've split this one out just to get a bit of discussion going on the best approach to cover these. 

So in this I've tried to make a more isolated test showing how to check conditionals in its own directory, along with a README to try to explain it a little. Any feedback on this would be appreciated before I progress with the other two I've got in progress (naming conventions & parameter naming).

shadycuz commented:
@dhutchison This is really nice, I think you should add something to the main README to point to the examples folder so its easy for people to find.

dhutchison commented:
Yep that should have been an obvious omission on my part, added a mention in now. 

Technically this is probably a minor change as opposed to patch as I've added an extra method to `Stack` for getting all resources of a type - so it's not just a documentation change anymore.

Going to mark this one as ready for review & move on to the next one.

shadycuz reviewed (approved):
Looks really good =), thanks for adding real documentation, my examples turned into some crazy edge case unit tests.

shadycuz commented:
@dhutchison Feel free to set the PR labels however you want. I can update the contributing guide if it's not clear or make a guide if we don't have one for this repo. But you need to add a type of change and the bump type. I think the pipeline has tests for the labels. The labels get picked up by https://github.com/release-drafter/release-drafter to create our automatic changelog and to create the Semver tags and the GitHub release automatically.

You can also clone this repo and create branches directly in it instead of using your fork. I don't mind if you use your fork but for some reason when we merge to develop, it won't give the pipeline access to the secret var that contains the Pypi credentials and the pipeline fails. I will have to figure out a fix for that, so you can use your fork and future contributors 🤞 won't have any issues. 

I also use "rebases" for feature branches into develop and merge commits for develop into main 🙄.

dhutchison commented:
I'll take a nosey at the pipeline later in the week to see if anything jumps out but I'm sure I've seen a similar issue in one of my repos with PRs from dependabot and never got the bottom of it. 

I didn't think of that, yep I'll start branching off the main repo for the next few ones. And when I'm my laptop I'll take a look but I didn't think it had let me set labels since I noticed the PR build failed on that - won't be until later on tomorrow.
